posthog/api/dashboard.py

```diff
@@ -161,7 +161,7 @@ def get_items(self, dashboard: Dashboard):
         if self.context["view"].action == "list":
             return None
 
-        items = dashboard.items.filter(deleted=False).order_by("order").all()
```

Comment:
This appears to be correct, but I'm not actually sure where the filtering _does_ place. Could you pinpoint?

Reply:
It's in `get_queryset` https://github.com/PostHog/posthog/blob/139bcae5b64a4465daac4a42c91d368ed1f4c6dd/posthog/api/dashboard.py#L223

Reply:
Hmm, I see. But does it hurt to leave the `.filter(deleted=False)` in this line? I'm just a bit afraid someone in the future will use the serializer in a place that doesn't have that in `.prefetch_related()`, resulting in soft-deleted insights showing up.

Reply:
🤔  reasonable concern....

It only adds 1 query, so now reduced from 13 to 9. I can live with that :)